defender/utils.py

```diff
@@ -243,15 +267,24 @@ def lockout_response(request):
 
 def is_already_locked(request):
     """ Is this IP/username already locked? """
-    ip_address = get_ip(request)
-    username = request.POST.get(config.USERNAME_FORM_FIELD, None)
 
-    # ip blocked?
-    ip_blocked = REDIS_SERVER.get(get_ip_blocked_cache_key(ip_address))
+    if not config.DISABLE_IP_LOCKOUT:
+        # ip blocked?
+        ip_address = get_ip(request)
+        ip_blocked = REDIS_SERVER.get(get_ip_blocked_cache_key(ip_address))
+    else:
+        # we disabled ip lockout, so it will never be blocked.
+        ip_blocked = False
 
     # username blocked?
+    username = request.POST.get(config.USERNAME_FORM_FIELD, None)
     user_blocked = REDIS_SERVER.get(get_username_blocked_cache_key(username))
 
+    if config.DISABLE_IP_LOCKOUT:
+        # if DISABLE_IP_LOCKOUT is turned on, then return user_blocked
+        # result, since we don't care about ip at this point.
+        return user_blocked
```

Comment:
I don't think we need this condition. Instead, on line 295, get rid of the big `else` clause, and just do `return ip_blocked or user_blocked`.

Btw, line 290 should just be `return ip_blocked and user_blocked`. No need to do an `if`.